### PR TITLE
[Notifier] Add iSendPro bridge

### PR DESCRIPTION
**Pull request by leblanc-simon**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/17450
|Recipe         | https://github.com/symfony/recipes/pull/1147

This PR add the notifier bridge for [Isendpro](https://www.isendpro.com/) (SMS).

Replace the #48253 (I failed to target new branch with Git, sorry...)

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    protected const HOST = 'apirest.isendpro.com';
+
+    public function __construct(
+        #[\SensitiveParameter] private string $keyid,
+        private ?string $from = null,
+        private bool $noStop = false,
+        private bool $sandbox = false,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php:36]: online please

File: src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (null === $this->from) {
+            return sprintf('isendpro://%s?no_stop=%d&sandbox=%d', $this->getEndpoint(), $this->noStop, $this->sandbox);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php]: noStop and sandbox are booleans and need an int cast

File: src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+        return sprintf('isendpro://%s?from=%s&no_stop=%d&sandbox=%d', $this->getEndpoint(), $this->from, $this->noStop, $this->sandbox);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php]: same here

File: src/Symfony/Component/Notifier/Bridge/Isendpro/composer.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=8.1",
+        "symfony/http-client": "^5.4|^6.0",
+        "symfony/notifier": "^6.3"
+    },
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Isendpro/composer.json:26]: you should add require-dev with symfony/event-dispatcher ^5.4|^6.0

File: src/Symfony/Component/Notifier/Bridge/Isendpro/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+See iSendPro documentation at https://www.isendpro.com/docs/#prerequis
+
+Resources
+---------
+
+* [Contributing](https://symfony.com/doc/current/contributing/index.html)
+* [Report issues](https://github.com/symfony/symfony/issues) and
+  [send Pull Requests](https://github.com/symfony/symfony/pulls)
+  in the [main Symfony repository](https://github.com/symfony/symfony)
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Isendpro/README.md]:
```suggestion
where:
 - `ACCOUNT_KEY_ID` is your iSendPro API Key ID
 - `FROM` is the alphanumeric originator for the message to appear to originate from  (optional)
 - `NO_STOP` setting this parameter to "1" (default "0") allows removing "STOP clause" at the end of the message for non-commercial use (optional)
 - `SANDBOX` setting this parameter to "1" (default "0") allows to use the notifier in sandbox mode (optional)

See iSendPro documentation at https://www.isendpro.com/docs/#prerequis

Resources
---------

 * [Contributing](https://symfony.com/doc/current/contributing/index.html)
 * [Report issues](https://github.com/symfony/symfony/issues) and
   [send Pull Requests](https://github.com/symfony/symfony/pulls)
   in the [main Symfony repository](https://github.com/symfony/symfony)
```

**Comment by leblanc-simon** [comment]
Thanks @OskarStark for your review !

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @leblanc-simon.

**Pull request merged**